modules/caddyhttp/reverseproxy/command.go

```diff
@@ -175,6 +177,13 @@ func cmdReverseProxy(fs caddycmd.Flags) (int, error) {
 	if fromAddr.Scheme == "http" {
 		server.AutoHTTPS = &caddyhttp.AutoHTTPSConfig{Disabled: true}
 	}
+	if disableRedir {
+		if server.AutoHTTPS != nil {
+			server.AutoHTTPS = &caddyhttp.AutoHTTPSConfig{DisableRedir: true}
```

Comment:
I don't like that it takes priority over `fromAddr.Scheme == "http"`

Opinions on throwing an error in such case instead? :eyes:

Reply:
@IndeedNotJames Thanks!
Thus, you mean `fromAddr.Scheme == “http”` takes priority over redirect disabled. 
How would you like that it ignores disable redirects if user specifies both of fromAddr.Scheme http and redirect disabled instead it occurs an error?

Reply:
Hi, sorry for the very vague comment yesterday.
I just had another look at the code, so let me try to clarify it:

I originally observed that
```
caddy reverse-proxy --from http://localhost:2020 --to https://example.com --change-host-header --disable-redirects
```

will start serving HTTPS on `localhost:2020` (despite having `--from http://localhost:2020`).

So I was like "if I, as a user, explicitly pass `http://` in `--from http://localhost:2020`, then I want Caddy to only listen on `localhost:2020`, serving HTTP".

And considering there are no HTTP->HTTPS redirects on an HTTP-only server, why allow `--disable-redirects` in the first place?

---

Now that I had a closer look at the code again, I noticed that

`--disable-redirects` takes priority over `&caddyhttp.AutoHTTPSConfig{Disabled: true}`, because you override the existing `AutoHTTPSConfig` struct, if `server.AutoHTTPS != nil`.
I think you meant `if server.AutoHTTPS == nil {` instead:

```diff
@@ -178,7 +178,7 @@ func cmdReverseProxy(fs caddycmd.Flags) (int, error) {
                server.AutoHTTPS = &caddyhttp.AutoHTTPSConfig{Disabled: true}
        }
        if disableRedir {
-               if server.AutoHTTPS != nil {
+               if server.AutoHTTPS == nil {
                        server.AutoHTTPS = &caddyhttp.AutoHTTPSConfig{DisableRedir: true}
                } else {
                        server.AutoHTTPS.DisableRedir = true
```

So that explains why `--from http://localhost:2020` with `--disable-redirects` enabled HTTPS on `localhost:2020`.

`--disable-redirects` took priority over `--from http://localhost:2020` due to a small little typo in `if server.AutoHTTPS == nil`

---

However, from the [AutoHTTPS json docs](https://caddyserver.com/docs/json/apps/http/servers/automatic_https/):

> [disable](https://caddyserver.com/docs/json/apps/http/servers/automatic_https/disable/)
> If true, automatic HTTPS will be entirely disabled, including certificate management and redirects.

So knowing that
```
&caddyhttp.AutoHTTPSConfig{DisableRedir: true}
```
does nothing if
```
&caddyhttp.AutoHTTPSConfig{Disabled: true}
```
is already set (because `{Disabled: true}` already disables all the AutoHTTPS features, including `{DisableRedir: true}`), then you could as well just write
```diff
@@ -176,13 +176,8 @@ func cmdReverseProxy(fs caddycmd.Flags) (int, error) {
 
        if fromAddr.Scheme == "http" {
                server.AutoHTTPS = &caddyhttp.AutoHTTPSConfig{Disabled: true}
-       }
-       if disableRedir {
-               if server.AutoHTTPS != nil {
-                       server.AutoHTTPS = &caddyhttp.AutoHTTPSConfig{DisableRedir: true}
-               } else {
-                       server.AutoHTTPS.DisableRedir = true
-               }
+       } else if disableRedir {
+               server.AutoHTTPS = &caddyhttp.AutoHTTPSConfig{DisableRedir: true}
        }
```

Which simplifies the code a bit.

---

So it's probably perfectly fine to allow `--disable-redirects` whenever `fromAddr.Scheme == "http"`.
No need to throw an error or whatever, since they don't actually conflict with each other :)

However, I would suggest (diff above) only setting `{DisableRedir: true}` when `{Disabled: true}` isn't already set :)